### ISSUE TITLE
Add `baseBranch` option in pre mode

### ISSUE THREAD

**Issue by Andarist** [changesets/changesets#234]
## Problem

With a rather popular workflow of having master & next branch (latter for pre mode) changesets targeting next branch have "incorrect" hints provided by `getChangedPackagesSinceRef`. Changed packages are computed as if that changeset was targeting a master branch and this prompt menu becomes way less useful because of that in pre mode.

## Proposed solution

When entering pre mode prompt for `baseBranch` to be configured, if it's given then it should shallow the "default" one for computing things like `getChangedPackagesSinceRef`.

I could work on adding this if we establish how this should work exactly.

**Comment by mitchellhamilton** [comment]
I think we could do this by making the pre enter command `changeset pre enter <tag> <baseBranch>` where baseBranch is optional and defaults to the normal baseBranch in the config

**Comment by Andarist** [comment]
Couldn't we prompt for it explicitly when entering pre mode (if not given through CLI arguments)? It seems like developing pre stuff on a separate branch is a pretty common use case and entering pre mode is not, so an extra step there shouldn't hurt anyone but would bring this issue to the user's attention.

**Comment by mitchellhamilton** [comment]
Yeah, I think prompting would be good. Though I think it should probably be `changeset pre enter` prompts for both the tag and branch. `changeset pre enter <tag>` uses the currently configured baseBranch so that the current usage doesn't change. `changeset pre enter <tag> <baseBranch>` uses both the arguments.

**Comment by Andarist** [comment]
Ok, I'll try to implement it like this when I get time - some higher priority stuff ahead of me right now though, so I'm unsure when I will be able to do so. If anyone wants to pick it up sooner - that would be fine with me as well.
